pkg/storage/s3.go

```diff
@@ -1,56 +1,288 @@
-	Endpoint string `json:"endpoint" toml:"endpoint"`
-	Region   string `json:"region" toml:"region"`
+	Endpoint              string `json:"endpoint" toml:"endpoint"`
+	Region                string `json:"region" toml:"region"`
+	StorageClass          string `json:"storage_class" toml:"storage_class"`
+	SSE                   string `json:"sse" toml:"sse"`
+	ACL                   string `json:"acl" toml:"acl"`
+	AccessKey             string `json:"access_key" toml:"access_key"`
+	SecretAccessKey       string `json:"secret_access_key" toml:"secret_access_key"`
+	Provider              string `json:"provider" toml:"provider"`
+	ForcePathStyle        bool   `json:"force_path_style" toml:"force_path_style"`
+	UseAccelerateEndpoint bool   `json:"use_accelerate_endpoint" toml:"use_accelerate_endpoint"`
 }
 
 func (options *S3BackendOptions) apply(s3 *backup.S3) error {
-	if options.Endpoint == "" && options.Region == "" {
-		return errors.New("must provide either 's3.region' or 's3.endpoint'")
+	if options.Region == "" {
+		options.Region = "us-east-1"
+	}
+	if options.Endpoint != "" {
+		u, err := url.Parse(options.Endpoint)
+		if err != nil {
+			return err
+		}
+		if u.Scheme == "" {
+			return errors.New("scheme not found in endpoint")
+		}
+		if u.Host == "" {
+			return errors.New("host not found in endpoint")
+		}
+	}
+	// In some cases, we need to set ForcePathStyle to false.
+	// Refer to: https://rclone.org/s3/#s3-force-path-style
+	if options.Provider == "alibaba" || options.Provider == "netease" ||
+		options.UseAccelerateEndpoint {
+		options.ForcePathStyle = false
+	}
+	if options.AccessKey == "" && options.SecretAccessKey != "" {
+		return errors.New("access_key not found")
+	}
+	if options.AccessKey != "" && options.SecretAccessKey == "" {
+		return errors.New("secret_access_key not found")
 	}
-	// TODO: Verify Region.
 
 	s3.Endpoint = options.Endpoint
 	s3.Region = options.Region
+	// StorageClass, SSE and ACL are acceptable to be empty
+	s3.StorageClass = options.StorageClass
+	s3.Sse = options.SSE
+	s3.Acl = options.ACL
+	s3.AccessKey = options.AccessKey
+	s3.SecretAccessKey = options.SecretAccessKey
+	s3.ForcePathStyle = options.ForcePathStyle
 	return nil
 }
 
 func defineS3Flags(flags *pflag.FlagSet) {
-	flags.String(s3EndpointOption, "", "Set the AWS S3 endpoint URL")
-	flags.String(s3RegionOption, "", "Set the AWS region")
-	// TODO: Finalize the list of options.
-	_ = flags.MarkHidden(s3EndpointOption)
+	flags.String(s3EndpointOption, "", "Set the S3 endpoint URL, please specify the http or https scheme explicitly")
+	flags.String(s3RegionOption, "", "Set the S3 region, e.g. us-east-1")
+	flags.String(s3StorageClassOption, "", "Set the S3 storage class, e.g. STANDARD")
+	flags.String(s3SSEOption, "", "Set the S3 server-side encryption algorithm, e.g. AES256")
+	flags.String(s3ACLOption, "", "Set the S3 canned ACLs, e.g. authenticated-read")
+	flags.String(s3ProviderOption, "", "Set the S3 provider, e.g. aws, alibaba, ceph")
```

Comment:
This setting doesn't seem to have any effect yet besides `force-path-style`. I suggest exposing `force-path-style` instead of this option, until the provider can really change the endpoint etc.